extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -72,16 +78,35 @@ class AmpYoutube extends AMP.BaseElement {
     iframe.width = this.width_;
     iframe.height = this.height_;
     this.element.appendChild(iframe);
-    /** @private {?Element} */
+
+    /** @private {!Element} */
     this.iframe_ = iframe;
 
-    // TODO(mkhatib, #2050): Use PlayerReady message for layout promise.
-    return loadPromise(iframe);
+    /** @private @const {!Promise} */
+    this.playerReadyPromise_ = this.getPlayerReadyPromise_();
+
+    this.getWin().addEventListener(
```

Comment:
Yeah you're right. Adding the `message` event handler to `getWin` in tests (which returns AMP_TEST window) doesn't seem to catch YT messages correctly. When I add it to the global window it does. 

I'll do some more debugging to see why that's the case.